enhancements/kubelet/split-filesystem.md

```diff
@@ -0,0 +1,282 @@
+
+#### Checking if feature is enabled on a node.
+
+One can use `crictl imagefsinfo` to see if the filesystem is split. This will show imageFilesystems and containerFilesystems.
+
+If they are split you would need a different mount in containerFilesystems.
+
+## Proposal
+
+### Background
+
+The main way to enable a split filesystem case is add `imageStore` in the container storage configuration.
+
+See [container storage](https://github.com/containers/storage/blob/main/docs/containers-storage.conf.5.md).
+
+```
+imagestore="" The image storage path (the default is assumed to be the same as graphroot). Path of the imagestore, which is different from graphroot. By default, images in the storage library are stored in the graphroot. If imagestore is provided, newly pulled images will be stored in the imagestore location. All other storage continues to be stored in the graphroot. When using the overlay driver, images previously stored in the graphroot remain accessible. Internally, the storage library mounts graphroot as an additionalImageStore to allow this behavior.
+```
+
+Container storage is configured in openshift by adding this [file](https://github.com/openshift/machine-config-operator/blob/master/templates/common/_base/files/container-storage.yaml) to /etc/containers/storage.conf.
+
+Container Runtime Config allows one to change the overlay size of storage. Other fields of this file are kept the same as the template.
+
+
+### Feature Gate
+
+We will add a feature gate to openshift/api.
+
+```golang
+	FeatureGateKubeletSeparateDiskGC = newFeatureGate("KubeletSeparateDiskGC").
+						reportProblemsToJiraComponent("node").
+						contactPerson("kannon92").
+						productScope(kubernetes).
+						enableIn(configv1.DevPreviewNoUpgrade).
+						mustRegister()
+```
+
+### Configuration of container storage
+
+### API Changes
+
+```golang
+type ContainerRuntimeConfiguration struct {
+  ...
+	// +optional
+	SplitFilesystem bool `json:"splitFilesystem,omitempty"`
+
+}
+```
+
+API is defined [here](https://github.com/openshift/api/blob/0d46442e8df17a87cfcf6666ab19b28e88620b59/machineconfiguration/v1/types.go#L812).
+
+This feature will write the updated container storage file.
+
+This will also trigger labeling of /var/lib/images.
+
+### Risks and Mitigations
+
+Kubernetes and Openshift do not really advertise the support of separate filesystems.
+We also do not allow for most configuration of the container runtime. Changing configuration
+in this area can break your system.
+
+To derisk this scenario, in tech preview, we will propose an API to configure imagestore.
+
+### Drawbacks
+
+## Test Plan
+
+TBD
+
+## Graduation Criteria
+
+### Dev Preview
+
+- Ability to view if a user has configured this feature.
+- Feature gates to enable this feature
+- API change to streamline configuration of this feature.
+
+### Dev Preview -> Tech Preview
+
+Will update once we are ready to promote.
```

Comment:
API Changes go here.

I also suggest other requirements:

* Installer Support
* Day-2 support for additional filesystems

This is where we call out external dependencies to make the feature better from a user's viewpoint.